core/services/chainlink/config_general_test.go

```diff
@@ -131,3 +127,131 @@ func TestConfig_LogSQL(t *testing.T) {
+		}
+		err = opts.Setup(configFiles, secretsFiles)
+		require.NoErrorf(t, err, "error: %s", err)
+
+		err = opts.parse()
+		require.NoErrorf(t, err, "error testing: %s, %s", configFiles, secretsFiles)
+
+		assert.Equal(t, databaseSecrets.Database.URL.URL().String(), opts.Secrets.Database.URL.URL().String())
+		assert.Equal(t, databaseSecrets.Database.BackupURL.URL().String(), opts.Secrets.Database.BackupURL.URL().String())
+
+		assert.Equal(t, (string)(*explorerSecrets.Explorer.AccessKey), (string)(*opts.Secrets.Explorer.AccessKey))
+		assert.Equal(t, (string)(*explorerSecrets.Explorer.Secret), (string)(*opts.Secrets.Explorer.Secret))
+		assert.Equal(t, (string)(*passwordSecrets.Password.Keystore), (string)(*opts.Secrets.Password.Keystore))
+		assert.Equal(t, (string)(*passwordSecrets.Password.VRF), (string)(*opts.Secrets.Password.VRF))
+		assert.Equal(t, (string)(*pyroscopeSecrets.Pyroscope.AuthToken), (string)(*opts.Secrets.Pyroscope.AuthToken))
+		assert.Equal(t, (string)(*prometheusSecrets.Prometheus.AuthToken), (string)(*opts.Secrets.Prometheus.AuthToken))
+		assert.Equal(t, (string)(*thresholdSecrets.Threshold.ThresholdKeyShare), (string)(*opts.Secrets.Threshold.ThresholdKeyShare))
+
+		err = assertDeepEqualityMercurySecrets(*merge(mercurySecrets_a.Mercury, mercurySecrets_b.Mercury), opts.Secrets.Mercury)
+		require.NoErrorf(t, err, "merged mercury secrets unequal")
+	})
+}
+
+func parseSecrets(secrets string) (*Secrets, error) {
+	var s Secrets
+	if err := config.DecodeTOML(strings.NewReader(secrets), &s); err != nil {
+		return nil, fmt.Errorf("failed to decode secrets TOML: %w", err)
+	}
+
+	return &s, nil
+}
+
+func assertDeepEqualityMercurySecrets(expected toml.MercurySecrets, actual toml.MercurySecrets) error {
+	if len(expected.Credentials) != len(actual.Credentials) {
+		return fmt.Errorf("maps are not equal in length: len(expected): %d, len(actual): %d", len(expected.Credentials), len(actual.Credentials))
+	}
+
+	for key, value := range expected.Credentials {
+		equal := true
+		actualValue := actual.Credentials[key]
+		if (string)(*value.Username) != (string)(*actualValue.Username) {
+			equal = false
+		}
+		if (string)(*value.Password) != (string)(*actualValue.Password) {
+			equal = false
+		}
+		if value.URL.URL().String() != actualValue.URL.URL().String() {
+			equal = false
+		}
+		if !equal {
+			return fmt.Errorf("maps are not equal: expected[%s] = {%s, %s, %s}, actual[%s] = {%s, %s, %s}",
+				key, (string)(*value.Username), (string)(*value.Password), value.URL.URL().String(),
+				key, (string)(*actualValue.Username), (string)(*actualValue.Password), actualValue.URL.URL().String())
+		}
+	}
+	return nil
+}
+
+func merge(map1 toml.MercurySecrets, map2 toml.MercurySecrets) *toml.MercurySecrets {
+	combinedMap := make(map[string]toml.MercuryCredentials)
+
+	for key, value := range map1.Credentials {
+		combinedMap[key] = value
+	}
+
+	for key, value := range map2.Credentials {
+		combinedMap[key] = value
+	}
```

Comment:
Could simplify this with `maps.Copy` https://pkg.go.dev/golang.org/x/exp/maps#Copy

Reply:
Thanks! Will do 💯